shared/utility/strategy/soccer_strategy.hpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2025 NUbots
+ *
+ * This file is part of the NUbots codebase.
+ * See https://github.com/NUbots/NUbots for further info.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef UTILITY_STRATEGY_SOCCER_STRATEGY_HPP
+#define UTILITY_STRATEGY_SOCCER_STRATEGY_HPP
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+#include "message/input/RoboCup.hpp"
+#include "message/localisation/Ball.hpp"
+#include "message/localisation/Field.hpp"
+#include "message/strategy/TeamMates.hpp"
+
+
+namespace utility::strategy {
+
+    using message::localisation::Ball;
+    using message::localisation::Field;
+    using message::strategy::TeamMates;
+
+    struct Possession {
+        enum Value { SELF, TEAMMATE, OPPONENT, NONE };
+        Value value = Value::SELF;
+    };
+
+    struct Robots {
+        // Robot's distance to ball
+        double distance_to_ball = 0.0;
+        // Id of the robot
+        uint id = 0;
+        // Instance of Possession struct, indicating if the robot has the ball, is close, etc.
+        Possession possession;
+    };
+
+    std::vector<Robots> get_sorted_bots(const Ball& ball, const TeamMates& teammates, const Field& field) {
```

Comment:
yeah this looks like you're going about it the right way by making this sorted list